spec/lib/oauth/client_credentials/creator_spec.rb

```diff
@@ -11,8 +11,35 @@ class Doorkeeper::OAuth::ClientCredentialsRequest
       end.to change { Doorkeeper::AccessToken.count }.by(1)
     end
 
+    context "when reuse_access_token is true" do
+      before do
+        allow(Doorkeeper.configuration).to receive(:reuse_access_token).and_return(true)
+      end
+
+      it 'returns the existing valid token when one exists' do
+        existing_token = subject.call(client, scopes)
+        existing_token_count = Doorkeeper::AccessToken.count
+        result = subject.call(client, scopes)
+        expect(Doorkeeper::AccessToken.count).to eq(existing_token_count)
+        expect(result).to eq(existing_token)
+      end
+    end
+
+    context "when reuse_access_token is false" do
+      before do
+        allow(Doorkeeper.configuration).to receive(:reuse_access_token).and_return(false)
+      end
+
+      it 'returns the existing valid token when one exists' do
+        subject.call(client, scopes)
+        expect do
+          subject.call(client, scopes)
+        end.to change { Doorkeeper::AccessToken.count }.by(1)
+      end
+    end
```

Comment:
I suggest the following rewrite of this spec:

```ruby
context "when reuse_access_token is true" do
  it "returns the existing valid token" do
    allow(Doorkeeper.configuration).to receive(:reuse_access_token).and_return(true)
    existing_token = subject.call(client, scopes)

    result = subject.call(client, scopes)

    expect(Doorkeeper::AccessToken.count).to eq(1)
    expect(result).to eq(existing_token)
  end
end

context "when reuse_access_token is false" do
  it "returns a new token" do
    allow(Doorkeeper.configuration).to receive(:reuse_access_token).and_return(false)
    existing_token = subject.call(client, scopes)

    result = subject.call(client, scopes)

    expect(Doorkeeper::AccessToken.count).to eq(2)
    expect(result).not_to eq(existing_token)
  end
end
```

* It follows the [setup/execute/assertions structure](https://robots.thoughtbot.com/four-phase-test), making it easier to read
* It removes the unneeded `before` block
* It makes both options look more alike, to better illustrate the difference (with the config on or off, it's easier to visualize how behavior changes)
* It corrects the description for the second test
* It introduces an explaining variable for `existing_token` in the second spec
* It consistently uses double quotes

Could you update your copy? It might not be correct syntax (don't have my tests set up right now), and Hound might have one or two complaints, but I'd like to keep these advantages.

Thank you!